src/Agda2Hs/Compile/Term.hs

```diff
@@ -272,7 +272,7 @@ specialClassFunction3 v f = specialClassFunction $ \case
   es               -> v `eApp` es
 
 fromNat :: ProjCompileRule
-fromNat = specialClassFunction1 (hsVar "fromIntegral") $ \case
+fromNat = specialClassFunction2 (hsVar "fromIntegral") $ \v _ -> case v of
```

Comment:
I trust that the code still works, but can you explain what requires using `specialClassFunction2` now?

I couldn't find where the signature of `fromIntegeral` got changed, thus where the extra argument comes from.

Reply:
The extra argument is the (instance) argument `{{_ : Constraint n}}` in the type of `fromNat`:

https://github.com/agda/agda/blob/0def25833eb92660499799280aa791f58253b4bb/src/data/lib/prim/Agda/Builtin/FromNat.agda#L11

Previously this was treated as a class constraint (i.e. checked for canonicity and then dropped) but now it becomes are regular argument instead so we need to drop it by hand. In an ideal world, we would update the type of `fromNat` to mark the constraint as erased, but it is defined in `Agda.Builtin` so changing it is not easy.

Reply:
Do you think defining our own `fromNat` and `Number` class is a bad idea?

Reply:
If you want to use literals on the Agda side then it *has* to use the `fromNat`/`fromNeg`/`fromString` functions from the builtin Agda modules.

Reply:
Well yes you can declare the `BUILTIN` yourself, but declaring builtins is something we've been avoiding so far to preserve compatibility with other libraries (at least for the erased parts of your `agda2hs` code).